### DISCUSSION TITLE
Loop Step for Automation to repeat the action over rows of a query?

### DISCUSSION THREAD

**Discussion by superbolt-x** [Budibase/budibase#4416] [Help] [answered]
Hi!

I would want to know if it's possible to have in the automation builder a way to query many rows (e.g. a list of slack usernames) and perform the action every loop step of the row - e.g.  1 Slack Action (send a slack) per row for all those rows. I guess using something like {{ trigger.info.i.email }} using an index i instead of the specific 0 or 1 etc

This would help for example in batching a list of people and run a cron automation to slack them only once a day in an easy way

Thanks!

**Comment by mjashanks** [comment, marked as answer]
We cannot do this at the moment - although I agree that this would be an extremely useful addition. So many uses open up for this... I've logged this as an enhancement here: https://github.com/Budibase/budibase/issues/4427

We will do this, but cant give a date.

**Reply by superbolt-x** [reply]
Gotcha! Thanks a lot!!!
